agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -4458,6 +4464,49 @@ bool backhaul_manager::handle_slave_channel_selection_response(ieee1905_1::CmduM
     return send_cmdu_to_bus(cmdu_tx, controller_bridge_mac, bridge_info.mac);
 }
 
+bool backhaul_manager::handle_backhaul_steering_request(ieee1905_1::CmduMessageRx &cmdu_rx,
+                                                        const std::string &src_mac)
+{
+    const auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received BACKHAUL_STA_STEERING message, mid=" << std::hex << int(mid);
+
+    auto bh_sta_steering_req = cmdu_rx.getClass<wfa_map::tlvBackhaulSteeringRequest>();
+    if (!bh_sta_steering_req) {
+        LOG(ERROR) << "Failed cmdu_rx.getClass<wfa_map::tlvBackhaulSteeringRequest>(), mid="
+                   << std::hex << int(mid);
+        return false;
+    }
+
+    auto channel = bh_sta_steering_req->target_channel_number();
+    auto bssid   = bh_sta_steering_req->target_bssid();
+
+    /*
+        TODO: Need to do channel validation. If channel invalid, send ACK and Backhaul STA Sterring
+              Response with the error code without trigerring HAL.
+    */
+
+    auto active_hal = get_wireless_hal();
+    if (!active_hal) {
+        LOG(ERROR) << "Couldn't get active HAL";
+        return false;
+    }
+
+    auto associate = active_hal->associate(bssid, channel);
```

Comment:
```suggestion
    auto associate = active_hal->roam(bssid, channel);
```

The proper way is probably to use the state machine instead. That said, maybe it's not needed, and it's way more efficient if we can roam directly here rather than going through the state machine.

Reply:
Roger that.